src/SonarScanner.MSBuild.TFS.Classic/Program.cs

```diff
@@ -0,0 +1,99 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2019 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using SonarScanner.MSBuild.Common;
+using SonarScanner.MSBuild.Common.Interfaces;
+using SonarScanner.MSBuild.TFS.Classic.XamlBuild;
+
+namespace SonarScanner.MSBuild.TFS.Classic
+{
+    public static class Program
+    {
+        public static int Main(string[] args)
+        {
+            var logger = new ConsoleLogger(includeTimestamp: false);
+            return Execute(args, logger);
+        }
+
+        public static int Execute(string[] args, ILogger logger)
+        {
+            try
+            {
+                if (args.Length < 1)
+                {
+                    logger.LogError("No argument found. Exiting...");
+                    Environment.ExitCode = 0;
+                    return 0;
+                }
+
+                var method = (Method)Enum.Parse(typeof(Method), args[0], false);
+                var legacyTeamBuildFactory = new LegacyTeamBuildFactory(logger);
+                var teamBuildSettings = TeamBuildSettings.GetSettingsFromEnvironment(logger);
+                AnalysisConfig config = null;
+
+                switch (method)
+                {
+                    case Method.ConvertCoverage:
+                        config = AnalysisConfig.Load(args[1]);
```

Comment:
If I'm reading the code correctly, the expected arguments for each mode are as follows:

* Coverage converter `[method] [config file path] [full properties file path]`
* Report builder: `[method] [ran to completion] [config file path] [full properties file path]`

It would be simpler if the common arguments were passed in the same locations for both commands, with the one extra argument (`ranToCompletion`) appearing last.

Reply:
You're right !